### ISSUE TITLE
Make the name of the system assembly "mscorlib" configurable

### ISSUE THREAD

**Issue by ChrML** [bridgedotnet/Bridge#4027]
**Problem summary**
Under .NET Framework the system assembly is named "mscorlib".
Under .NET Core the system assembly is named "System.Private.CoreLib".

When serializing data on a .NET Core server with fully-qualified you get typenames such as "System.String, System.Private.CoreLib". It causes problems for the client finding the types which is expecting to find "System.String, mscorlib". And vice-versa.

**Suggestion**
Make the Bridge.NET system library assembly configurable in bridge.json. Perhaps a setting like:

```
{
  // Choose Framework or Core- typenaming.
  "outputSystemAssemblyFormat": "Core/Framework"
}
```

### Steps To Reproduce
https://deck.net/91eb7aa6ce7b02df49f5ede1a29790d9

```csharp
public class Program
{
    public static void Main()
    {
        // Get the fully qualified name for System.String
        string typeName = typeof(String).AssemblyQualifiedName;
        Console.WriteLine(typeName);
        
        // Outputs: System.String, mscorlib
    }
}
```

### Expected Result

```
System.String, mscorlib
// or
System.String, System.Private.CoreLib
```

### Actual Result

```
System.String, mscorlib
```

**Comment by fabriciomurta** [comment]
Hi ChrML, but I suppose aside of that, there's a long way for Bridge to make it to .NET Core, I don't really get how you're getting Bridge to work under net core.

I believe that once Bridge -supports- .NET Core, it could be able to just use the corresponding string given where it was built (`mscorlib` if .NET Framework, `System.Private.CoreLib` if .NET Core).

**Comment by ChrML** [comment]
Hi.

For me, only the server is .NET Core. The client is still Framework, files are only served by the Core server. The problem now is mainly the interoperability with the server.

Not a huge problem though, was able to resolve it using some find/replace hacks before/after serialization.

**Comment by fabriciomurta** [comment]
Got it. Makes sense!
Like this, one may even set up to have queries to a .NET Core endpoint and also other queries to a .NET Framework endpoint, which suggests in this (edge) case, the setting would be desirable at the query/serialization level. Or am I still off tracks here?

**Comment by ChrML** [comment]
@fabriciomurta Yeah, you got it! Even after Bridge is ported to Core, one may see Framework endpoints.

I'm not sure how one would go about configuring this at serialization/endpoint level though.

**Comment by fabriciomurta** [comment]
One possible way I think is a nonstandard [Newtonsoft.Json.JsonSerializerSettings](https://github.com/bridgedotnet/Bridge.Newtonsoft.Json/blob/master/Newtonsoft.Json/Json/JsonSerializerSettings.cs) to flag how `mscorlib` type gets turned into string, and maybe something else to map either `mscorlib`/`System.Private.Corelib` back into `mscorlib` on deserialization.

Now, I'm just not sure we could just drive the library off stock syntax.

**Comment by ChrML** [comment]
I'm not sure that's the best way. It will likely cause some problems down the road, by deviating from the standard library.

I think the best way is ignoring the special case where you might communicate with multiple API endpoints, where one may be Core and another Framework. In most cases there is only one endpoint. In some cases it may be multiple using Cross-Origin, but in these cases I think the user should either adapt the endpoints or hack these situations manually.

I think the best way is introducing a global setting for the entire project, at compiler level. It could have 3 settings: "auto/core/framework". Where Auto it would choose the library native to how Bridge is compiled (only framework for now).

I'm not sure if the mscorlib assembly name is coded at Bridge level, or if it comes from a deeper level (eg Roslyn). In case it is coded at Bridge level, then a selection would be required in the future anyway to make it System.Private.CoreLib when compiled on Core.

Another reason for implementing this at compiler level is cases where the user shares code between the client and the server using Shared Projects in Visual Studio. Ensuring that the assemblies have the same name on both sides, increases the likelyhood of the code working identical and correctly both at the client (Bridge) and the server.

**Comment by fabriciomurta** [comment]
Yes, great points, can't see how I could agree more with that. :)
